src/lib/net/internal/index.ts

```diff
@@ -0,0 +1,107 @@
+import { Players, RunService } from "@rbxts/services";
+
+import { EventListener } from "../../events";
+import { FunctionUtil } from "../../util/functions";
+import { Identifiers } from "../util/identifiers";
+import { ClientListener, Packet, ServerListener } from "../util/types";
+import { InternalNetworkingEvents } from "./types";
+
+export namespace Internals {
+	let remote: RemoteEvent | undefined;
+
+	const listeners: { [x: string]: Array<ClientListener | ServerListener> } = {};
+	const isServer = RunService.IsServer();
+
+	const queue: Array<{ to?: Array<Player>; contents?: object; id: string }> = [];
+
+	// eslint-disable-next-line no-inner-declarations
+	function listen(id: string, callback: ClientListener | ServerListener): void {
+		if (listeners[id] === undefined) {
+			listeners[id] = [callback];
+		} else {
+			listeners[id].push(callback);
+		}
+	}
+
+	/** @hidden */
+	export function init(): void {
+		if (!isServer) {
+			remote = script.WaitForChild("[Internals]") as RemoteEvent | undefined;
+
+			RunService.PostSimulation.Connect(() => {
+				remote?.FireServer(queue);
```

Comment:
`remote!.FireServer...`

Reply:
Yes and no, it may find itself erroring if it doesn't find it, which I don't want 60 errors popping up every second.